db/seeds/votes.rb

```diff
@@ -8,13 +8,16 @@
   "votes.csv"
 )
 
+rows = []
+
 CSV.foreach(votes_filename, headers: true, encoding: "BOM|UTF-8") do |row|
-  params = {
+  rows << {
+    id: row["id"].to_i,
     place_id: row["place"],
     uuid: row["uuid"],
     rating: row["rating"],
     created_at: DateTime.parse(row["date_submitted"])
   }
-
-  Vote.find_or_create_by(id: row["id"].to_i).update!(params)
 end
+
+Vote.upsert_all(rows.uniq { |vote| [vote[:place_id], vote[:uuid]] }, unique_by: :id)
```

Comment:
If I understand this right, we're eliminating any accidental duplicate votes in Ruby, before we even pass them to the `upsert`?

Reply:
Yeah, that's right 👍

Reply:
I've made that into a variable of its own to make that clearer now